digitalocean/dropletautoscale/resource_droplet_autoscale.go

```diff
@@ -0,0 +1,375 @@
+				sshKeys = append(sshKeys, key.(string))
+			}
+		}
+
+		return &godo.DropletAutoscaleResourceTemplate{
+			Size:             poolTemplate["size"].(string),
+			Region:           poolTemplate["region"].(string),
+			Image:            poolTemplate["image"].(string),
+			Tags:             tags,
+			SSHKeys:          sshKeys,
+			VpcUUID:          poolTemplate["vpc_uuid"].(string),
+			WithDropletAgent: poolTemplate["with_droplet_agent"].(bool),
+			ProjectID:        poolTemplate["project_id"].(string),
+			IPV6:             poolTemplate["ipv6"].(bool),
+			UserData:         poolTemplate["user_data"].(string),
+		}
+	}
+	return nil
+}
+
+func resourceDigitalOceanDropletAutoscaleCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	pool, _, err := client.DropletAutoscale.Create(context.Background(), &godo.DropletAutoscalePoolRequest{
+		Name:            d.Get("name").(string),
+		Config:          expandConfig(d.Get("config").([]interface{})),
+		DropletTemplate: expandTemplate(d.Get("droplet_template").([]interface{})),
+	})
+	if err != nil {
+		return diag.Errorf("Error creating Droplet autoscale pool: %v", err)
+	}
+	d.SetId(pool.ID)
+
+	// Setup to poll for autoscale pool scaling up to the desired count
+	stateConf := &retry.StateChangeConf{
+		Delay:      5 * time.Second,
+		Pending:    []string{"provisioning"},
+		Target:     []string{"active"},
+		Refresh:    dropletAutoscaleRefreshFunc(client, d.Id()),
+		MinTimeout: 15 * time.Second,
+		Timeout:    15 * time.Minute,
+	}
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return diag.Errorf("Error waiting for Droplet autoscale pool (%s) to become active: %v", pool.Name, err)
+	}
+
+	return resourceDigitalOceanDropletAutoscaleRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanDropletAutoscaleRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	pool, _, err := client.DropletAutoscale.Get(context.Background(), d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), fmt.Sprintf("autoscale group with id %s not found", d.Id())) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error retrieving Droplet autoscale pool: %v", err)
+	}
+
+	d.Set("name", pool.Name)
+	d.Set("config", flattenConfig(pool.Config))
+	d.Set("current_utilization", flattenUtilization(pool.CurrentUtilization))
+	d.Set("status", pool.Status)
+	d.Set("created_at", pool.CreatedAt.UTC().String())
+	d.Set("updated_at", pool.UpdatedAt.UTC().String())
+
+	// Persist existing image specification (id/slug) if it exists
+	if t, ok := d.GetOk("droplet_template"); ok {
+		tList := t.([]interface{})
+		if len(tList) > 0 {
+			tMap := tList[0].(map[string]interface{})
+			if tMap["image"] != "" {
+				pool.DropletTemplate.Image = tMap["image"].(string)
+			}
+		}
+	}
+	d.Set("droplet_template", flattenTemplate(pool.DropletTemplate))
```

Comment:
cc @andrewsomething opted for this change lmk what you think: we're basically trying to avoid any in-place updates essentially.